spec/actions_spec.rb

```diff
@@ -232,12 +253,113 @@ def self.name
     end
 
     it 'deletes a non-existing record' do
-      instance = klass.new(value: 1)
+      instance = klass.new(value: 1, indexed_value: 1)
       expect(instance.id).to be_nil
       expect {
         instance.destroy
       }.to_not raise_error
       expect(klass.count).to eq 0
     end
   end
+
+  context 'custom indexes: ' do
+    let!(:time_now) { Time.zone.now}
+    let!(:instance) { klass.create!(indexed_value: 1, time_value: time_now) }
+    let!(:instance_2) { klass.create!(indexed_value: 2, time_value: nil) }
+
+    it 'creates an attribute in index content hash for custom index' do
+      unless ENV['REDCORD_SPEC_USE_CLUSTER'] == 'true'
+        index_string = klass.redis.hget("#{klass.model_key}:custom_index_first_content", instance.id)
+        expect(index_string).to_not be(nil)
+      end
+    end
+
+    it 'returns instance by int attribute query' do
+      expect(klass.where(indexed_value: 1, index: :first).to_a.first.id).to eq(instance.id)
+    end
+
+    it 'returns count by int attribute query' do
+      expect(klass.where(indexed_value: 1, index: :first).count).to eq(1)
+      expect(klass.where(indexed_value: 3, index: :first).count).to eq(0)
+    end
+
+    it 'returns instance by time attribute range query' do
+      interval = Redcord::RangeInterval.new(min: time_now - 10.seconds)
+      if ENV['REDCORD_SPEC_USE_CLUSTER'] == 'true'
+        expect {
+          klass.where(time_value: interval, index: :second).to_a
+        }.to raise_error(Redcord::AttributeNotIndexed)
+      else
+        expect(klass.where(time_value: interval, index: :second).to_a.first.id).to eq(instance.id)
+      end
+    end
+
+    it 'returns instance by attribute is nil query' do
+      if ENV['REDCORD_SPEC_USE_CLUSTER'] == 'true'
+        expect {
+          klass.where(time_value: nil, index: :second).to_a
+        }.to raise_error(Redcord::AttributeNotIndexed)
```

Comment:
can we make sure we also run some valid test cases with cluster mode on?

Reply:
sure, added some